include/fastdds/rtps/security/logging/Logging.h

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Canonical ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file Logging.h
+ */
+#ifndef _FASTDDS_RTPS_SECURITY_LOGGING_LOGGING_H_
+#define _FASTDDS_RTPS_SECURITY_LOGGING_LOGGING_H_
+
+#include "fastdds/rtps/security/logging/LogOptions.h"
+#include "fastdds/rtps/security/exceptions/SecurityException.h"
+
+namespace eprosima {
+namespace fastrtps {
+namespace rtps {
+namespace security {
+
+/**
+ * @brief The LoggerListener class
+ */
+class LoggerListener
+{
+  LoggerListener() = default;
+  ~LoggerListener() = default;
+};
+
+/**
+ * @brief Base class for all security logging plugins.
+ */
+class Logging
+{
+public:
+
+  Logging();
+  virtual ~Logging() = default;
+
+  /**
+   * @brief set_log_options
+   * @param log_options
+   * @return TRUE if successful
+   */
+  bool set_log_options(const LogOptions& log_options, SecurityException& exception);
+
+  /**
+   * @brief get_log_options
+   * @param log_options
+   * @param exception
+   * @return
+   */
+  bool get_log_options(LogOptions& log_options, SecurityException& exception) const;
+
+  /**
+   * @brief enable_logging
+   */
+  bool enable_logging(SecurityException& exception);
+
+  /**
+   * @brief set_listener
+   * @param listener
+   * @return
+   */
+  bool set_listener(LoggerListener* listener, SecurityException& exception);
+
+  /**
+   * @brief log
+   * @param log_level
+   * @param message
+   * @param category
+   */
+  void log(const EventLogLevel event_log_level,
```

Comment:
Should all of these public methods be final?

Reply:
Since their behavior and their logic relative to one another (e.g. set_option before log) are clearly defined in the specs I'd argue that they should.
Note that `log` internally calls `log_impl`, after the logic bits, which allows for modifying the implementation details.

Reply:
Indeed, and that's perfect. If we make them final it makes it abundantly clear how subclasses are supposed to work (just reimplement this one little function...).